packages/dds/tree/src/core/tree/visitDelta.ts

```diff
@@ -409,10 +440,9 @@ function detachPass(delta: Delta.FieldChanges, visitor: DeltaVisitor, config: Pa
 			}
 			if (isDetachMark(mark)) {
 				for (let i = 0; i < mark.count; i += 1) {
-					const root = config.detachedFieldIndex.createEntry(
-						// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-						offsetDetachId(mark.detach!, i),
-					);
+					// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+					const id = offsetDetachId(mark.detach!, i);
+					const root = config.detachedFieldIndex.createEntry(id, config.latestRevision);
```

Comment:
We need test visitDelta test coverage to check that the latest revision gets set.